### PR TITLE
Create U2F device registration form

### PR DESCRIPTION
**Pull request by rjkip**
- [x] Merge #1

### CODE REVIEW

File: src/Form/Type/RegisterDeviceType.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Surfnet\StepupU2fBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class RegisterDeviceType extends AbstractType
```
Comment (DRvanR) [inline src/Form/Type/RegisterDeviceType.php:27]: please shuffle the methods to have an order that is a bit more expected :wink:

File: src/Resources/public/u2f.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            registerRequest = $form.data('u2f-register-request');
+
+        u2f.register([registerRequest], [], function (response) {
+            $errorCode.val(response.errorCode || ERROR_CODE_OK);
+            $registrationData.val(response.registrationData);
+            $clientData.val(response.clientData);
+            $form.submit();
+        });
+    });
+
```
Comment (DRvanR) [inline src/Resources/public/u2f.js:25]: an empty line?

File: src/Service/RegistrationVerificationResult.php [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            RegisterResponse::ERROR_CODE_OK,
+            RegisterResponse::ERROR_CODE_OTHER_ERROR,
+            RegisterResponse::ERROR_CODE_BAD_REQUEST,
+            RegisterResponse::ERROR_CODE_CONFIGURATION_UNSUPPORTED,
+            RegisterResponse::ERROR_CODE_DEVICE_INELIGIBLE,
+            RegisterResponse::ERROR_CODE_TIMEOUT,
+        ];
+
+        if (!in_array($errorCode, $validErrorCodes, true)) {
+            throw new DomainException('Device error code is not one of the known error codes');
```
Comment (DRvanR) [inline src/Service/RegistrationVerificationResult.php]: mwwwwyyyeaaaah, not so sure about `DomainException` here...

**Comment by DRvanR** [comment]
Few small things, looking good otherwise!

File: src/Form/Type/RegisterDeviceType.php (hunk 1 above)
Comment (rjkip) [inline src/Form/Type/RegisterDeviceType.php:27]: What is a more expected order? :( `getName`, `buildForm`, the rest?

File: src/Resources/public/u2f.js (hunk 2 above)
Comment (rjkip) [inline src/Resources/public/u2f.js:25]: Correct.

Comment (DRvanR) [inline src/Resources/public/u2f.js:25]: Nice :dancers:

**Comment by rjkip** [comment]
From: @rjkip
To: @DRvanR 
Subject: Shipment requested

I request shipment.

**Comment by DRvanR** [comment]
:+1: , :ship: it!

**Pull request merged**